src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; }[] }>(requestOptions)
+      .then(response => {
+        const group: { id: string; } | undefined = response.value[0];
+
+        if (!group) {
+          return Promise.reject(`The specified owner group does not exist`);
+        }
+
+        return Promise.resolve(group.id);
+      });
+  }
+
+  private getUserIds(logger: Logger, users: string): Promise<string[]> {
+    const userArr: string[] = users.split(',').map(o => o.trim());
+    let promises: Promise<{ value: User[] }>[] = [];
+    let userIds: string[] = [];
+
+    promises = userArr.map(user => {
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/users?$filter=userPrincipalName eq '${Utils.encodeQueryParameter(user)}'&$select=id,userPrincipalName`,
+        headers: {
+          'content-type': 'application/json'
+        },
+        responseType: 'json'
+      };
+
+      return request.get(requestOptions);
+    });
+
+    return Promise.all(promises).then((usersRes: { value: User[] }[]): Promise<string[]> => {
+      let userUpns: string[] = [];
+
+      userUpns = usersRes.map(res => res.value[0]?.userPrincipalName as string);
+      userIds = usersRes.map(res => res.value[0]?.id as string);
+
+      // Find the members where no graph response was found
+      const invalidUsers = userArr.filter(user => !userUpns.some((upn) => upn?.toLowerCase() === user.toLowerCase()));
+
+      if (invalidUsers && invalidUsers.length > 0) {
+        return Promise.reject(`Cannot proceed with planner task creation. The following users provided are invalid : ${invalidUsers.join(',')}`);
+      }
+      return Promise.resolve(userIds);
+    });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-t, --title <title>'
+      },
+      {
+        option: "--planId [planId]"
+      },
+      {
+        option: "--planName [planName]"
+      },
+      {
+        option: "--ownerGroupId [ownerGroupId]"
+      },
+      {
+        option: "--ownerGroupName [ownerGroupName]"
+      },
+      {
+        option: "--bucketId [bucketId]"
+      },
+      {
+        option: "--bucketName [bucketName]"
+      },
+      {
+        option: "--startDateTime [startDateTime]"
+      },
+      {
+        option: "--dueDateTime [dueDateTime]"
+      },
+      {
+        option: "--percentComplete [percentComplete]"
+      },
+      {
+        option: "--assignedToUserIds [assignedToUserIds]"
+      },
+      {
+        option: "--assignedToUserNames [assignedToUserNames]"
+      },
+      {
+        option: "--description [description]"
+      },
+      {
+        option: "--orderHint [orderHint]"
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.title) {
```

Comment:
We don't need to check if required options have been set. This is done centrally in the CLI, so we can remove this piece.